testdata/baselines/reference/submodule/conformance/jsDeclarationsMissingGenerics.js.diff

```diff
@@ -10,4 +10,27 @@
   * @param {Promise} x
   */
 -declare function y(x: Promise<any>): void;
-+declare function y(x: Promise): void;
\ No newline at end of file
++declare function y(x: Promise): void;
++
++
++//// [DtsFileErrors]
++
++
++out/file.d.ts(4,23): error TS2314: Generic type 'Array<T>' requires 1 type argument(s).
```

Comment:
Hm, this is exciting; wonder how easy it'd be to fix this?

Reply:
@weswigham can correct me, but I think this needs `symbolTableToDeclarationStatements` and others before we expect JS `.d.ts` to *really* work - see `existingTypeNodeIsNotReferenceOrIsReferenceWithCompatibleTypeArgumentCount`, `canReuseTypeNode` etc.

Reply:
Actually, not so - since we changed how JS code is parsed, we may _never_ implement the `symbolTableToDeclarationStatements` stub and might just remove it (which is maybe a problem for expandable hover, since that also leverages it). Since JS is syntactically transformed into something the checker likes now, similarly, the declaration emitter aughta just have those special syntaxes handled directly in declaration emit rather than always falling back to a semantic printback - which basically just boils down to handling the new reparsed node structures and kinds in the transform itself. I added a draft of whatever js support was in while I was working on the initial declaration emitter, but it really does just need someone to go in and add the entirely new emit logic.

Reply:
In this case, the declaration emitter is picking up that we now just shove the type nodes directly on the `.Type` of the parameter and mostly just doing the right thing, but the declaration emitter is missing the logic to ensure minimum type argument counts are met that the node builder used to handle implicitly. That's a slightly semantic question ("how many type arguments should this reference have, at a minimum?"), so it'll probably need to get answered through a new `emitResolver` hook.